opts/quotedstring_test.go

```diff
@@ -16,15 +16,19 @@ func TestQuotedStringSetWithQuotes(t *testing.T) {
 }
 
 func TestQuotedStringSetWithMismatchedQuotes(t *testing.T) {
-	value := ""
-	qs := NewQuotedString(&value)
+	qs := NewQuotedString(new(string))
```

Comment:
Orthogonal to this PR, but I just noticed that `QuotedString` only performs the quote stripping when using `Set()`, but `NewQuotedString()` does not do the same, so theoretically, if `value` was a string with quotes, nothing would happen, and `.String()` would return the quoted value as-is. https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/opts/quotedstring.go#L34-L37


I _think_ that was an oversight (at least, I can't come up with good reasons to have it not do the same). Anyway, I think it's fine to fix that separately